src/common/systems/economy/sysfactory.cpp

```diff
@@ -30,114 +30,120 @@
-                       cqspc::Market& market) {
+                    cqspc::Market& market) {
     // Get the transport cost
     auto& infrastructure =
         universe.get<cqspc::infrastructure::CityInfrastructure>(entity);
     // Calculate the infrastructure cost
     double infra_cost =
         infrastructure.default_purchase_cost - infrastructure.improvement;
 
-    auto& industries = universe.get<cqspc::Industry>(entity);
+    auto& industries = universe.get<cqspc::IndustrialZone>(entity);
     for (entt::entity productionentity : industries.industries) {
         // Process imdustries
         // Industries MUST have production and a linked recipe
         if (!universe.all_of<components::Production>(productionentity))
             continue;
-        components::Recipe recipe = universe.get_or_emplace<components::Recipe>(
-            universe.get<components::Production>(productionentity).recipe);
-        components::FactorySize& size =
-            universe.get_or_emplace<components::FactorySize>(productionentity,
-                                                             1000.0);
-        components::ProductionRatio& ratio =
-            universe.get_or_emplace<components::ProductionRatio>(
-                productionentity, 1.);
-
+        components::Recipe recipe =
+            universe.get_or_emplace<components::Recipe>(
+                universe.get<components::Production>(productionentity)
+                    .recipe);
+        components::IndustrySize& size =
+            universe.get_or_emplace<components::IndustrySize>(
+                productionentity, 1000.0);
+        // Calculate resource consumption
+        components::ResourceLedger capitalinput =
+            recipe.capitalcost * (0.01 * size.size);
         components::ResourceLedger input =
-            (recipe.input * ratio.ratio) +
-            (recipe.capitalcost * (0.01 * size.size));
-        // If there is not enough input goods, then restrict the trades
-        // Input
-        double input_transport_cost = input.GetSum() * infra_cost;
+            (recipe.input + size.size) + capitalinput;
 
+        // Calculate the greatest possible production
         components::ResourceLedger output;  // * ratio.output;
-        output[recipe.output.entity] = recipe.output.amount * ratio.ratio;
-        double output_transport_cost = output.GetSum() * infra_cost;
+        output[recipe.output.entity] =
+            recipe.output.amount * size.size;
 
-        // Check demand if there is demand
-        if (market.previous_demand[recipe.output.entity] > 0) {
-            // Then they can sell it
-            market.supply[recipe.output.entity] += recipe.output.amount * ratio.ratio;
-        } else {
-            // IDK what to do
-        }
+        // Figure out what's throttling production and maintaince
+        double limitedinput =
+            CopyVals(input, market.history.back().sd_ratio).Min();
```

Comment:
Is there a way to not use CopyVals? Can you make it more efficient by not copying the maps so many times?